src/EntryPoints/PersistentPageIdentifiersHooks.php

```diff
@@ -53,4 +53,20 @@ public static function onPageSaveComplete(
 		PersistentPageIdentifiersExtension::getInstance()->newCreatePersistentPageIdentifier()->createId( $wikiPage->getId() );
 	}
 
+	public static function onRevisionUndeleted( RevisionRecord $restoredRevision, ?int $oldPageId ): void {
+		if ( $oldPageId !== null ) {
+			return;
+		}
+
+		// TODO: untested
+		$repo = PersistentPageIdentifiersExtension::getInstance()->getPersistentPageIdentifiersRepo();
+		$oldPersistentId = $repo->getPersistentId( $restoredRevision->getPageId() );
+
+		if ( $oldPersistentId === null ) {
+			return;
+		}
+
+		$repo->savePersistentId( $restoredRevision->getPageId(), $oldPersistentId );
```

Comment:
This code is incomplete anyway, since we'll need to remove the old row in `persistent_page_ids` table due to the UNIQUE constraint.

Otherwise, we can drop the constraint, but then we end up with different `page_id`s mapping to the same `persistent_id`. This seems fundamentally wrong:
* there should be only one page with a specific persistent ID
* if deleted page_id=1 has persistent_id=FOO, and an undelete leads to it becoming page_id=2 with peristent_id=FOO, would it still be possible to undelete page_id=1 again?

But the above is dependent on figuring out when a page ID is missing in the first place.